File: list_primes.go
package main

import (
	"fmt"
	"math"
)

func isPrime(number int) bool {
	if number <= 1 {
		return false
	} else if number == 2 {
		return true
	} else if number%2 == 0 {
		return false
	}

	loop_limit := int(math.Floor(math.Sqrt(float64(number))))

	for i := 3; i <= loop_limit; i++ {
		if number%i == 0 {
			return false
		}
	}

	return true
}

func main() {
	for i := 1; i <= 500000; i++ {
		if isPrime(i) {
			fmt.Println(i)
		}
	}
}

File: list-primes.js
function isPrime(number) {
  if (number <= 1) {
    return false;
  } else if (number === 2) {
    return true;
  } else if (number % 2 === 0) {
    return false;
  }

  const loopLimit = Math.floor(Math.sqrt(number));

  for (let i = 3; i <= loopLimit; i = i + 2) {
    if (number % i === 0) {
      return false;
    }
  }

  return true;
}

const main = () => {
  for (let number = 1; number <= 500000; number++) {
    if (isPrime(number)) {
      console.log(number);
    }
  }
};

main();
